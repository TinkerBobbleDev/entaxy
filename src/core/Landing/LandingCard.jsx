import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import { Link } from 'react-router-dom'
import Card from '@material-ui/core/Card'
import CardHeader from '@material-ui/core/CardHeader'
import IconButton from '@material-ui/core/IconButton'
import ArrowForward from '@material-ui/icons/ArrowForward'

const styles = {
  card: {
    'margin-top': '20px',
    'flex-grow': 1
  },
  cardButton: {
    'margin-top': '10px'
  }
}

const LandingCard = ({
  classes,
  title,
  description,
  path
}) => (
  <Card className={classes.card}>
    <CardHeader
      title={title}
      subheader={description}
      action={
        <IconButton
          variant="fab"
          color="secondary"
          className={classes.cardButton}
          component={Link}
          to={path}
        >
          <ArrowForward />
        </IconButton>
      }
    />
  </Card>
)

LandingCard.propTypes = {
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired
}

export default withStyles(styles)(LandingCard)
