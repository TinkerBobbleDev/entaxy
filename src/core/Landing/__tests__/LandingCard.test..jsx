import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import LandingCard from '../LandingCard'

describe('LandingCard', () => {
  it('matches snapshot', () => {
    const component = renderer.create((
      <BrowserRouter>
        <LandingCard title="title" description="description" path="/taxes" />
      </BrowserRouter>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
