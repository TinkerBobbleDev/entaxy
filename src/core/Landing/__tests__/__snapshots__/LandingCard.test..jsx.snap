// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingCard matches snapshot 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 MuiCard-root-3 LandingCard-card-1"
>
  <div
    className="MuiCardHeader-root-31"
  >
    <div
      className="MuiCardHeader-content-34"
    >
      <span
        className="MuiTypography-root-37 MuiTypography-headline-42 MuiCardHeader-title-35"
      >
        title
      </span>
      <span
        className="MuiTypography-root-37 MuiTypography-body1-46 MuiTypography-colorTextSecondary-59 MuiCardHeader-subheader-36"
      >
        description
      </span>
    </div>
    <div
      className="MuiCardHeader-action-33"
    >
      <a
        className="MuiButtonBase-root-67 MuiIconButton-root-61 MuiIconButton-colorSecondary-64 LandingCard-cardButton-2"
        href="/taxes"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
        variant="fab"
      >
        <span
          className="MuiIconButton-label-66"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-70"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-76"
        />
      </a>
    </div>
  </div>
</div>
`;
